libbeat/processors/add_cloud_metadata/provider_aws_ec2.go

```diff
@@ -18,125 +18,134 @@
+		result.err = fmt.Errorf("failed loading AWS default configuration: %w", err)
+		return
 	}
+	awsClient := NewIMDSClient(awsConfig)
 
-	if rsp.StatusCode != http.StatusOK {
-		logger.Warnf("error when check request status for getting IMDSv2 token: http request status %d. No token in the metadata request will be used.", rsp.StatusCode)
-		return ""
+	instanceIdentity, err := awsClient.GetInstanceIdentityDocument(context.TODO(), &imds.GetInstanceIdentityDocumentInput{})
+	if err != nil {
+		logger.Debugf("error fetching EC2 Identity Document: %s.", err)
+		result.err = fmt.Errorf("failed fetching EC2 Identity Document: %w", err)
+		return
 	}
 
-	all, err := ioutil.ReadAll(rsp.Body)
+	// AWS Region must be set to be able to get EC2 Tags
+	awsRegion := instanceIdentity.InstanceIdentityDocument.Region
+	awsConfig.Region = awsRegion
+
+	clusterName, err := fetchEC2ClusterNameTag(awsConfig, instanceIdentity.InstanceIdentityDocument.InstanceID)
 	if err != nil {
-		logger.Warnf("error when reading token request for getting IMDSv2 token: %s. No token in the metadata request will be used.", err)
-		return ""
+		logger.Warnf("error fetching cluster name metadata: %s.", err)
 	}
 
-	return string(all)
-}
+	accountID := instanceIdentity.InstanceIdentityDocument.AccountID
 
-// AWS EC2 Metadata Service
-var ec2MetadataFetcher = provider{
-	Name: "aws-ec2",
+	_, _ = result.metadata.Put("instance.id", instanceIdentity.InstanceIdentityDocument.InstanceID)
+	_, _ = result.metadata.Put("machine.type", instanceIdentity.InstanceIdentityDocument.InstanceType)
+	_, _ = result.metadata.Put("region", awsRegion)
+	_, _ = result.metadata.Put("availability_zone", instanceIdentity.InstanceIdentityDocument.AvailabilityZone)
+	_, _ = result.metadata.Put("account.id", accountID)
+	_, _ = result.metadata.Put("image.id", instanceIdentity.InstanceIdentityDocument.ImageID)
 
-	Local: true,
+	// for AWS cluster ID is used cluster ARN: arn:partition:service:region:account-id:resource-type/resource-id, example:
+	// arn:aws:eks:us-east-2:627286350134:cluster/cluster-name
+	if clusterName != "" {
+		clusterARN := fmt.Sprintf("arn:aws:eks:%s:%s:cluster/%v", awsRegion, accountID, clusterName)
 
-	Create: func(_ string, config *conf.C) (metadataFetcher, error) {
-		ec2Schema := func(m map[string]interface{}) mapstr.M {
-			m["serviceName"] = "EC2"
-			out, _ := s.Schema{
-				"instance":          s.Object{"id": c.Str("instanceId")},
-				"machine":           s.Object{"type": c.Str("instanceType")},
-				"region":            c.Str("region"),
-				"availability_zone": c.Str("availabilityZone"),
-				"service": s.Object{
-					"name": c.Str("serviceName"),
-				},
-				"account": s.Object{"id": c.Str("accountId")},
-				"image":   s.Object{"id": c.Str("imageId")},
-			}.Apply(m)
-			return mapstr.M{"cloud": out}
-		}
+		_, _ = result.metadata.Put("orchestrator.cluster.name", clusterName)
+		_, _ = result.metadata.Put("orchestrator.cluster.id", clusterARN)
```

Comment:
this field is already defined in ecs [orchestrator.cluster.id](https://www.elastic.co/guide/en/ecs/current/ecs-orchestrator.html#field-orchestrator-cluster-id)